regression/contracts/assigns_replace_06/test.desc

```diff
@@ -1,9 +1,9 @@
 CORE
 main.c
 --replace-all-calls-with-contracts
-^EXIT=0$
+^EXIT=10$
 ^SIGNAL=0$
-^VERIFICATION SUCCESSFUL$
+^VERIFICATION FAILED$
```

Comment:
This change looks scary in a "refactoring" PR. Why was this not caught before? ...

Reply:
Before, we were not havocing `array[]` correctly and this test description was wrong. Take a look at a snippet from this test case
```c
void foo(char c[]) __CPROVER_assigns(c)
{
}

void bar(char d[]) __CPROVER_assigns(d)
{
}

int main()
{
  char b[10];
  b[0] = 'a';
  b[1] = 'b';
  b[2] = 'c';
  b[3] = 'd';
  b[4] = 'e';
  b[5] = 'f';
  b[6] = 'g';
  b[7] = 'h';
  b[8] = 'i';
  b[9] = 'j';
  foo(b);
  assert(b[0] == 'a');
  /* ... */
```

During replacement, this should clearly fail. We still need to add more checks (on a separate PR), to ensure we are also havoc the contents of arrays correctly (see https://github.com/diffblue/cbmc/issues/6235).

Reply:
( Could this test case make it into one of those PRs please? )

Reply:
Yes @martin-cs I added a test case to the GitHub issue, so I can make sure it gets into the PR that fixes it.